internal/sync/syncer.go

```diff
@@ -335,3 +351,41 @@ func toUnstructuredAndApplyAnnotation(manifests []string, gitSyncName string) ([
 	}
 	return uns, nil
 }
+
+// updateCommitStatus will update the commit status in git sync CR also if
+// error happened while syncing the target state then it update the error
+// reason as well.
+func updateCommitStatus(
+	ctx context.Context,
+	kubeClient client.Client,
+	logger *zap.SugaredLogger,
+	namespacedName types.NamespacedName,
+	hash string,
+	synced bool,
+	errMsg string,
+) error {
+	commitStatus := v1alpha1.CommitStatus{
+		Hash:     hash,
+		Synced:   synced,
+		SyncTime: metav1.NewTime(time.Now()),
+		Error:    errMsg,
+	}
+
+	gitSync := &v1alpha1.GitSync{}
+	if err := kubeClient.Get(ctx, namespacedName, gitSync); err != nil {
+		// if we aren't able to do a Get, then either it's been deleted in the past, or something else went wrong
+		if apierrors.IsNotFound(err) {
+			return nil
+		} else {
+			logger.Errorw("Unable to get GitSync", "err", err)
+			return err
+		}
+	}
+
+	gitSync.Status.CommitStatus = &commitStatus
+	if err := kubeClient.Status().Update(ctx, gitSync); err != nil {
```

Comment:
so, I guess in the case of failing to update either because of network issue with K8S API, or resourceVersion conflict, we probably won't retry ever, right?

Reply:
If we did do exponential backoff here, it would only tie up one worker goroutine, right? and should be rare? What are your thoughts?

Reply:
I think it is fine to not retry with exponential backoff here as it will be picked up by another worker in next round. I don't see in numaflow we are doing retry if status update failed.